app/models/package_manager/go.rb

```diff
@@ -53,34 +53,54 @@ def self.update(name)
       # example: github.com/myexample/modulename/v2
       if (matches = name.match(VERSION_MODULE_REGEX))
         PackageManagerDownloadWorker.perform_async(self.name, matches[1])
+
+        module_project = Project.find_by(platform: "Go", name: name)
+        base_module_project = Project.find_by(platform: "Go", name: matches[1])
+        return if module_project.nil? || base_module_project.nil?
+
+        module_project.versions.each do |version|
+          base_module_project.versions.find_or_create_by(number: version[:number]) do |new_version|
+            new_version.update(published_at: version[:published_at])
+            new_version.save
+          end
```

Comment:
This'll only get run for new records, but if you wanted to run it for either new-or-existing, you could use `tap` instead:

```suggestion
          base_module_project.versions.find_or_create_by(number: version[:number]).tap do |new_version|
            new_version.update(published_at: version[:published_at])
          end
```

Reply:
I think the intent was to not worry about the versions we've already captured in the past. As an optimization we could even do something like:
```
        new_base_versions = module_project.versions - base_module_project.versions.where(id: module_project.versions).pluck(:number)
        new_base_versions.each do |version|
          base_module_project.versions.create(number: version[:number], published_at: version[:published_at])
        end
```